### PR TITLE
Add PremiumProgressBarEnabled to DiscordGuild.cs

### PR DESCRIPTION
**Pull request by OoLunar**
Testing:
```cs
using DSharpPlus.CommandsNext;
using DSharpPlus.CommandsNext.Attributes;
using System.Threading.Tasks;

namespace Tomoe.Commands.Test
{
    [Group("dsp_fix"), RequireOwner]
    public class Test : BaseCommandModule
    {
        [Command("boost_bar_enabled"), Description("Test command.")]
        public async Task BoostBarEnabled(CommandContext context) => await context.RespondAsync(context.Guild.PremiumProgressBarEnabled.ToString());
    }
}
```
image

On guild update, in the same session:
image

### CODE REVIEW

**Comment by OoLunar** [comment]
I partially apologize for my formatter. It seems to respect your formatting rules, except it enforces it's own rules when you don't define them in the `.editorconfig`

File: DSharpPlus/Entities/Guild/DiscordGuild.cs [hunk 1]
```diff
@@ -460,6 +460,12 @@ public string BannerUrl
         [JsonProperty("premium_subscription_count", NullValueHandling = NullValueHandling.Ignore)]
         public int? PremiumSubscriptionCount { get; internal set; }
 
+        /// <summary>
+        /// Whether the guild has the boost progress bar enabled.
+        /// </summary>
+        [JsonProperty("premium_progress_bar_enabled", NullValueHandling = NullValueHandling.Include)]
```
Comment (VelvetThePanda) [inline DSharpPlus/Entities/Guild/DiscordGuild.cs:466]: NVH include won’t do anything on a non-nullable struct :b

**Comment by Giggitybyte** [comment]
The diff for this PR makes me cry.

File: DSharpPlus/Entities/Guild/DiscordGuild.cs (hunk 1 above)
Comment (OoLunar) [reply DSharpPlus/Entities/Guild/DiscordGuild.cs:466]: Whoops, that's left over from debugging. Was meant to be `Ignore` as it was on the `NSFW` property

**Review by VelvetThePanda — COMMENTED** [review]
LGTM, but I think it’s preferable you try to keep unrelated reformatting out the PR

**Comment by OoLunar** [comment]
> LGTM, but I think it’s preferable you try to keep unrelated reformatting out the PR

Noted for the future

File: DSharpPlus/Entities/Guild/DiscordGuild.cs (hunk 1 above)
Comment (OoLunar) [reply DSharpPlus/Entities/Guild/DiscordGuild.cs]: Fixed in latest commit

**Comment by VelvetThePanda** [comment]
Thanks!

**Pull request merged**